qt/widgets/common/src/FitScriptGeneratorDataTable.cpp

```diff
@@ -115,32 +122,57 @@ FitScriptGeneratorDataTable::FitScriptGeneratorDataTable(QWidget *parent)
 
   connect(this, SIGNAL(itemClicked(QTableWidgetItem *)), this,
           SLOT(handleItemClicked(QTableWidgetItem *)));
+  connect(this, SIGNAL(itemSelectionChanged()), this,
+          SLOT(handleItemSelectionChanged()));
+  disconnect(this->verticalHeader(), SIGNAL(sectionPressed(int)), this,
+             SLOT(selectRow(int)));
 }
 
 bool FitScriptGeneratorDataTable::eventFilter(QObject *widget, QEvent *event) {
   if (widget == this->viewport()) {
     auto index = hoveredRowIndex(event);
 
-    if (index != m_lastIndex) {
-      if (this->item(m_lastIndex.row(), m_lastIndex.column()))
+    if (index != m_lastHoveredIndex) {
+      if (this->item(m_lastHoveredIndex.row(), m_lastHoveredIndex.column()))
         emit itemExited(index.isValid() ? index.row() : -1);
-      m_lastIndex = QPersistentModelIndex(index);
+      m_lastHoveredIndex = QPersistentModelIndex(index);
     }
   }
   return QTableWidget::eventFilter(widget, event);
 }
 
 void FitScriptGeneratorDataTable::handleItemClicked(QTableWidgetItem *item) {
-  m_selectedRow = item->row();
+  m_selectedRows = selectedRows();
   m_selectedColumn = item->column();
   if (m_selectedColumn == ColumnIndex::StartX ||
       m_selectedColumn == ColumnIndex::EndX)
     m_selectedValue = item->text().toDouble();
 }
 
+void FitScriptGeneratorDataTable::handleItemSelectionChanged() {
+  auto *selectionModel = this->selectionModel();
+
+  if (!selectionModel->hasSelection()) {
```

Comment:
could you explain what the purpose of this if is? I am a bit confused

Reply:
This is making sure that all the relevant rows which are selected during multi-selection get stored within the selectionModel. There was an edge case where this wasn't working as expected.

I'll add a comment to make this clearer.